tensorboard/components/tf_backend/test/requestManagerTests.ts

```diff
@@ -289,6 +289,81 @@ describe('backend', () => {
         // resolving the first request triggers finishTheTest
       });
     });
+
+    it('throws InvalidRequestOptionsError on bad RequestOptions', function() {
+      const rm = new RequestManager();
+        const badOptions = new RequestOptions();
+      badOptions.methodType = HttpMethodType.GET;
+      badOptions.body = "a body";
+      chai.assert.throws(
+        ()=>rm.requestWithOptions("http://www.google.com", badOptions),
+        InvalidRequestOptionsError);
+    });
+
+    describe('tests using sinon.fakeServer', function() {
+      let server;
+      
+      beforeEach(function() {
+        server = sinon.fakeServer.create();
+        server.respondImmediately = true;
+        server.respondWith("{}");
+      });
+
+      afterEach(function() {
+        server.restore();
+      });
+      
+      it('builds correct XMLHttpRequest when request(url) is called',
+         function() {
+           const rm = new RequestManager();
+           return rm.request("my_url")
+             .then(()=>{
+               chai.assert.lengthOf(server.requests, 1);
+               chai.assert.equal(server.requests[0].url, "my_url");
+               chai.assert.equal(server.requests[0].requestBody, null);
+               chai.assert.equal(server.requests[0].method, HttpMethodType.GET);
+               chai.assert.notProperty(server.requests[0].requestHeaders,
+                                       "Content-Type");
+             });
+         });
+
+      it('builds correct XMLHttpRequest when request(url, postData) is called',
+         function() {
+           const rm = new RequestManager();
+           return rm.request("my_url",
+                              {"key1": "value1", "key2": "value2"})
+             .then(() => {
+               chai.assert.lengthOf(server.requests, 1);
+               chai.assert.equal(server.requests[0].url, "my_url");
+               chai.assert.equal(server.requests[0].method,
+                                 HttpMethodType.POST);
+               chai.assert.instanceOf(server.requests[0].requestBody, FormData);
+               chai.assert.sameDeepMembers(
+                 Array.from((server.requests[0].requestBody as any).entries()),
```

Comment:
does a `FormData` not have `entries` according to TS? That's odd :(

Reply:
`FormData` does have `entries`, and `server` is typed as `any` anyway.
This cast is not required.